### DISCUSSION TITLE
Need to define a Model to get a count value?

### DISCUSSION THREAD

**Discussion by alper** [SeaQL/sea-orm#1088] [Q&A]
I look here: https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select/#handling-custom-selects

And it seems I need to create a struct to extract a count?

That seems a bit cluttered. My query just does a COUNT on the database and I would like to get the value out as an integer.

```rust
    let count_query = Venues::find()
        .filter(entity::venues::Column::GooglePlaceId.eq(""))
        .column_as(entity::venues::Column::Id.count(), "count")
        .all(sqlite_conn)
        .await
        .unwrap();
```

**Comment by alper** [comment]
I'm seeing this but that does not look any simpler: https://www.sea-ql.org/blog/#better-ergonomic-when-working-with-custom-select-list

It uses `into_values()` which here is undocumented: https://docs.rs/sea-orm/latest/sea_orm/entity/prelude/struct.Select.html#method.into_values

**Comment by alper** [comment]
Adding this does not work: `.into_values::<_, i64>()`

**Reply by billy1624** [reply]
Hey @alper, take this as an example.

```rs
#[derive(Copy, Clone, Debug, EnumIter, DeriveColumn)]
enum QueryAs {
    CakeName,
    NumOfCakes,
}

let res: Vec<(String, i64)> = cake::Entity::find()
    .select_only()
    .column_as(cake::Column::Name, QueryAs::CakeName)
    .column_as(cake::Column::Id.count(), QueryAs::NumOfCakes)
    .group_by(cake::Column::Name)
    .into_values::<_, QueryAs>()
    .all(&db)
    .await?;
```
